src/jubjub/mod.rs

```diff
@@ -125,13 +129,24 @@ pub trait JubjubParams<E: JubjubEngine>: Sized {
     /// Returns the window size for exponentiation of Pedersen hash generators
     /// outside the circuit
     fn pedersen_hash_exp_window_size(&self) -> u32;
+
+    fn exp_table_path(&self) -> &Option<String>;
+
+    fn file_backed_exp_table(&self) -> bool {
+        self.exp_table_path().is_some()
+    }
 }
 
 impl JubjubEngine for Bls12 {
     type Fs = self::fs::Fs;
     type Params = JubjubBls12;
 }
 
+fn edwards_d() -> Fr {
+    Fr::from_str("19257038036680949359750312669786877991949435402254120286184196891950884077233")
```

Comment:
this should be a constant, probably using `lazy_static`, no need to parse this str everytime it is used